From efd1373e7c4a7a89609184aa0b5e1c3edcb8be6a Mon Sep 17 00:00:00 2001
From: Ernesto Corona <ernesto.corona@intel.com>
Date: Fri, 19 Nov 2021 09:11:03 -0800
Subject: [PATCH] ASD Fix sprintf_s compilation issue for safec 3.5.1

A missing macro was defined to ensure sprintf_s macro defintion is
set when safe_str_lib.h is included.

Test:
Remove openbmc-meta-intel/meta-common/recipes-core/safec/safec_3.4.bb
Compile at-scale-debug using devtool
ASD Sanity finished successfully(SPR).

Signed-off-by: Ernesto Corona <ernesto.corona@intel.com>
Change-Id: Idd94b4ba9ff8b4194317a045b140ce90db2ee860
---
 CMakeLists.txt | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 59ff03c..d6239a4 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -4,6 +4,8 @@ project(at-scale-debug C)
 find_package (PkgConfig REQUIRED)
 include_directories(${CMAKE_CURRENT_SOURCE_DIR}/sysroot/include)
 pkg_check_modules (SAFEC REQUIRED libsafec)
+# Define HAVE_C99 to include sprintf_s macro in safec library
+set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -DHAVE_C99")
 include_directories (${SAFEC_INCLUDE_DIRS})
 link_directories (${SAFEC_LIBRARY_DIRS})
 
-- 
2.25.1

